src/Dialogs/BlockMultilinePasteDialog.vala

```diff
@@ -0,0 +1,50 @@
+/*
+* Copyright 2011-2018 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License version 3 as published by the Free Software Foundation.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+public class Terminal.BlockMultilinePasteDialog : Granite.MessageDialog {
+    public BlockMultilinePasteDialog (MainWindow parent) {
+        Object (
+            buttons: Gtk.ButtonsType.NONE,
+            transient_for: parent
+        );
+    }
+
+    construct {
+        image_icon = new ThemedIcon ("dialog-warning");
+        primary_text = _("The pasted text may execute multiple commands");
+
+        secondary_text =
```

Comment:
Shouldn't construct translatable text like this, if possible.  Suggest using `printf`.  Also there is a grammar error.
```
        secondary_text =  "%s\n%s".printf (
              _("The text you want to paste may be treated as a command and could unintentionally be executed."),
              _("Be sure your paste doesn't contain newline characters.")
        );
```